File: unitazIO/pooper.py
import time
import random

from .unitaz import Unitaz
from .pooping_exceptions import TooMuchTimeForPooping


class Poop(Unitaz):
    def __init__(self, poop_times: int, flush_after_pooping: bool = False) -> None:
        self.poop_times = poop_times
        self.flush_after_pooping = flush_after_pooping


    def boom(self, timer: int) -> None:
        if timer not in range(1, 1001):
            raise TooMuchTimeForPooping('Вы не можете какать меньше 1 или дольше 1000 секунд.')

        words_dict = ['БУУМ!', 'ТЫ ПРОСТО В УДАРЕ!', '#ТАКМНОГО!']
        
        for _ in range(self.poop_times):
            self.iteratively_pooped()
            time.sleep(timer)
            print(f'{" " * random.randint(5, 30)} {random.choice(words_dict)}')

        if self.flush_after_pooping:
            print('\nЯ все смыл, и на всякий случай вызвал сантехника.')

File: unitazTS/pooper.ts
import { Unitaz } from './unitaz';

export class Poop extends Unitaz {
    private poop_times: number;
    private flush_after_pooping: boolean = false;

    constructor(poop_times: number, flush_after_pooping: boolean = false) {
        super();

        this.poop_times = poop_times;
        
        if (flush_after_pooping) {
            this.flush_after_pooping = true;
        }
    }

    private randomNumber(from: number, to: number): number {
        const min = Math.ceil(from);
        const max = Math.floor(to);
        return Math.floor(Math.random() * (max - min + 1)) + min;
    }

    private spaces(): string {
        return ' '.repeat(this.randomNumber(5, 30));
    }

    public boom(timer: number): any {
        if (timer <= 0 || timer >= 101) {
            throw new Error('Вы не можете какать меньше 1 или дольше 1000 секунд.');
        }

        const wordsDict = ['БУУМ!', 'ТЫ ПРОСТО В УДАРЕ!', '#ТАКМНОГО!'];

        const wrapper = () => {
            this.iterativelyPooped()

            const randomIndex = Math.floor(Math.random() * wordsDict.length);
            const word = wordsDict[randomIndex];
            console.log(this.spaces() + word);
        };
        
        setInterval(wrapper, timer * 1000);
        
        if (this.flush_after_pooping) {
            console.log('\nЯ все смыл, и на всякий случай вызвал сантехника.');
        }
    }
}
